src/main/cms/cms_menu_ledstrip.c

```diff
@@ -39,51 +39,88 @@
 
 #include "fc/config.h"
 
+#include "io/ledstrip.h"
+
+#include "interface/settings.h"
 
 #ifdef USE_LED_STRIP
 
 static bool featureRead = false;
 static uint8_t cmsx_FeatureLedstrip;
+#ifdef USE_LED_PROFILES
+static uint8_t cmsx_LedProfile;
+static uint8_t cmsx_RaceColor;
+const char * const ledProfileNames[LED_PROFILE_COUNT] = {
+    "RACE",
+    "BEACON",
+#ifdef USE_LED_STRIP_STATUS_MODE
+    "STATUS"
+#endif
+};
+#endif
 
-static long cmsx_Ledstrip_FeatureRead(void)
+static long cmsx_Ledstrip_OnEnter(void)
 {
     if (!featureRead) {
         cmsx_FeatureLedstrip = featureIsEnabled(FEATURE_LED_STRIP) ? 1 : 0;
         featureRead = true;
     }
 
+#ifdef USE_LED_PROFILES
+    cmsx_LedProfile = getLedProfile();
+    cmsx_RaceColor = getLedRaceColor();
+#endif
     return 0;
 }
 
-static long cmsx_Ledstrip_FeatureWriteback(const OSD_Entry *self)
+static long cmsx_Ledstrip_OnExit(const OSD_Entry *self)
 {
     UNUSED(self);
     if (featureRead) {
         if (cmsx_FeatureLedstrip)
             featureEnable(FEATURE_LED_STRIP);
-        else
+        else {
+            ledStripDisable();
             featureDisable(FEATURE_LED_STRIP);
+        }
     }
 
+#ifdef USE_LED_PROFILES
+    setLedProfile(cmsx_LedProfile);
+    setLedRaceColor(cmsx_RaceColor);
+#endif
+
     return 0;
 }
 
+#ifdef USE_LED_PROFILES
 static OSD_Entry cmsx_menuLedstripEntries[] =
 {
     { "-- LED STRIP --", OME_Label, NULL, NULL, 0 },
     { "ENABLED",         OME_Bool,  NULL, &cmsx_FeatureLedstrip, 0 },
-
```

Comment:
the #ifdef should be around here and not the whole cmsx_menuLedstripEntries.

this current approach does not scale/work when e.g another #ifdef gets added. since there need to be a branch #if define(USE_LED_PROFILES) && define(USE_LED_STHNEW).. then one for each case and one for when both are undefined.

which will be hard to maintain since a code change to cmsx_menuLedstripEntries need to be done at multply places then. which anyone should avoid in any case

Reply:
Good point - thanks, I will change it.